lib/base/journaldlogger.cpp

```diff
@@ -0,0 +1,85 @@
+/* Icinga 2 | (c) 2021 Icinga GmbH | GPLv2+ */
+
+#if !defined(_WIN32) && defined(HAVE_SYSTEMD)
+#include "base/journaldlogger.hpp"
+#include "base/journaldlogger-ti.cpp"
+#include "base/configtype.hpp"
+#include "base/statsfunction.hpp"
+#include "base/sysloglogger.hpp"
+#ifndef SD_JOURNAL_SUPPRESS_LOCATION
+#define SD_JOURNAL_SUPPRESS_LOCATION
+#endif
```

Comment:
I'd rather add an `add_definitions(-DSD_JOURNAL_SUPPRESS_LOCATION)` there: https://github.com/Icinga/icinga2/blob/838406e105cd53d58f0ec7fbdbec4abfe5eab773/lib/base/CMakeLists.txt#L123-L128

Reasoning: if anything (indirectly) included before would also include `sd-journal.h`, the `#define` here would become ineffective. Should not be the case right now, but I think doing it in CMake would be more robust.

Reply:
Good point. How about adding the flag less globally, like
```cpp
set_source_files_properties(journaldlogger.cpp PROPERTIES COMPILE_FLAGS -DSD_JOURNAL_SUPPRESS_LOCATION)
```
Because, in case there's another usage of `systemd/sd-journal.h` somewhere, that other function may or may not want to suppress adding `CODE_*`, who knows :)

Reply:
If this turns to work out with our [custom unity builds](https://github.com/Icinga/icinga2/tree/95cdc00ad4e1ab788794322c92c54cd3f5c8c2bf/tools/mkunity), sure. But it seems to support [a property to disable it for individual files](https://github.com/Icinga/icinga2/blob/95cdc00ad4e1ab788794322c92c54cd3f5c8c2bf/tools/mkunity/CMakeLists.txt#L3-L8), so that might work. Otherwise, I'd go the path of least resistance.

(Yes, I know, CMake now has support for unity builds itself, but I'm not sure if for long enough that this would work on all the distros we build for.)

Reply:
Ah yes, I didn't consider the unity build. `EXCLUDE_UNITY_BUILD` didn't work out of the box, so let's stick with `add_definitions` for now.